darshan-util/pydarshan/darshan/log_utils.py

```diff
@@ -0,0 +1,94 @@
+"""
+Module for log handling utilities.
+"""
+
+import platform
+python_version = platform.python_version_tuple()
+# shim for convenient Python 3.9 importlib.resources
+# interface
+if int(python_version[1]) < 9 and int(python_version[0]) == 3:
+    import importlib_resources
+else:
+    # see: https://github.com/python/mypy/issues/1153
+    import importlib.resources as importlib_resources # type: ignore
+
+import os
+import glob
+import pathlib
+import pytest
+from typing import Union, Any
+
+try:
+    import darshan_logs
+    has_log_repo = True
+except ImportError:
+    has_log_repo = False
+
+
+def _locate_local_log(filename: str) -> Union[str, None]:
+    """Locates a log if available in `tests/input` or `examples/example-logs`."""
+    pydarshan_path = pathlib.Path(__file__).parent.parent
```

Comment:
It is best to avoid using `__file__` if possible (i.e., for reasons described in https://stackoverflow.com/a/58941536/2942522; I think I cheated and used it in a few places already, but it is sub-optimal). Plus the logs repo approach below is already using the desirable `importlib.resources` so matching that would make sense.

If that could be done, it would also enable more code reuse (see below). There is a big problem though--we can't actually use the data files from the `darshan` project install proper, because those files are never installed (!), which is one point I don't see discussed here or in the matching issue, so we wouldn't actually be unchaining ourselves from the need to run the testsuite from within a specific path in the local clone of the project.

It may be worth discussing with the team whether it would be acceptable to place the small number of files to be retained in the main repo in a standard installation for example. The drawback is likely size, which is probably why Jakob didn't include them by default.

```diff
--- a/darshan-util/pydarshan/darshan/log_utils.py
+++ b/darshan-util/pydarshan/darshan/log_utils.py
@@ -25,22 +25,9 @@ except ImportError:
     has_log_repo = False
 
 
-def _locate_local_log(filename: str) -> Union[str, None]:
-    """Locates a log if available in `tests/input` or `examples/example-logs`."""
-    pydarshan_path = pathlib.Path(__file__).parent.parent
-    local_dirs = ["tests/input", "examples/example-logs"]
-    for ldir in local_dirs:
-        ldir = os.path.join(pydarshan_path, ldir)
-        for log_path in glob.glob(os.path.join(ldir, "*.darshan")):
-            if filename in log_path:
-                return log_path
-    # if log is not found
-    return None
-
-
-def _locate_repo_log(filename: str) -> Union[str, None]:
-    """Locates a log in the darshan_logs repo."""
-    p = importlib_resources.files('darshan_logs') # type: Any
+def _locate_log(filename: str, project: str) -> Union[str, None]:
+    """Locates a log file in a project."""
+    p = importlib_resources.files(project) # type: Any
     darshan_logs_paths = [str(p) for p in p.glob('**/*.darshan')]
     for log_path in darshan_logs_paths:
         if filename in log_path:
@@ -74,12 +61,12 @@ def get_log_path(filename: str) -> str:
 
     """
     # try local paths
-    log_path = _locate_local_log(filename=filename)
+    log_path = _locate_log(filename=filename, project='darshan')
     if log_path:
         return log_path
     # try the logs repo
     if has_log_repo:
-        log_path = _locate_repo_log(filename=filename)
+        log_path = _locate_log(filename=filename, project='darshan_logs')
         if log_path:
             return log_path
         else:

```

Reply:
We could also cheat a bit by modifying the above to combine both approaches (always using `importlib` and also moving up a few dirs so that we can scan for the adjacent data dirs), which doesn't avoid the need for a local clone of `darshan` but does reduce duplication:

```diff
-def _locate_repo_log(filename: str) -> Union[str, None]:
-    """Locates a log in the darshan_logs repo."""
-    p = importlib_resources.files('darshan_logs') # type: Any
+def _locate_log(filename: str, project: str) -> Union[str, None]:
+    """Locates a log file in a project."""
+    p = importlib_resources.files(project).parent.parent # type: Any
     darshan_logs_paths = [str(p) for p in p.glob('**/*.darshan')]
```

That seems to "just work", but it still makes me cringe a bit---moving up a few dirs relative to a given target package could go searching for other paths/logs we don't want to be picking up.

Reply:
(for now, could only move up the dirs if `project` == `darshan`, which would just be a more concise version of the original logic)

Reply:
Thanks for the recommendations, I went ahead and simplified the log locating functions down and removed the dependence on `pathlib`. 

I think the remaining concern here is moving up 2 directories, but I haven't figured out another way of doing this  that doesn't involve moving the files themselves. Based on our meeting yesterday it sounds like that is a possibility, so maybe that will have to be fixed before this is merged.